rotkehlchen/chain/evm/transactions.py

```diff
@@ -541,6 +550,53 @@ def ensure_tx_data_exists(
         tx_receipt = self.dbevmtx.get_receipt(cursor, tx_hash, self.evm_inquirer.chain_id)
         return tx_data, tx_receipt  # type: ignore  # tx_data can't be None here
 
+    def get_and_ensure_internal_txns_of_parent_in_db(
+            self,
+            tx_hash: 'EVMTxHash',
+            to_address: 'ChecksumEvmAddress',
+            for_address: 'ChecksumEvmAddress',
+    ) -> list[EvmInternalTransaction]:
+        """Queries the internal transactions for an address of a parent hash where receiver is
+        to_address, saves them in the DB and returns them.
+
+        May raise:
+        - RemoteError if there is a problem querying the data sources or transaction hash does
+        not exist."""
+        db_internal_txs = self.dbevmtx.get_evm_internal_transactions(
+            parent_tx_hash=tx_hash,
+            blockchain=CHAINID_TO_SUPPORTED_BLOCKCHAIN[self.evm_inquirer.chain_id],
+        )
+        if len(db_internal_txs) > 0:
+            return db_internal_txs
```

Comment:
shouldn't you check the from and to here? so it is `len(checked_transactions) > 0` Otherwise I see that you could have internal transaction that could beling to an address that you are not looking for

Reply:
This function returns all of them. Filtering them with from/to in BaseDecoderTools